query/math.go

```diff
@@ -30,6 +30,16 @@ type mathTree struct {
 	Child []*mathTree
 }
 
+var (
+	ErrorIntOverflow     = errors.New("Integer overflow")
+	ErrorFloatOverflow   = errors.New("Float overflow")
+	ErrorDivisionByZero  = errors.New("Division by zero")
+	ErrorModuloByZero    = errors.New("Modulo by zero")
+	ErrorFractionalPower = errors.New("Fractional power of negative number")
+	ErrorNegativeLog     = errors.New("Log of negative number")
+	ErrorNegativeRoot    = errors.New("Root of negative number")
```

Comment:
All of these operations promote integer values to float, AFAIK. If we're returning floats here, we can just return `NaN` instead of an error.

Reply:
As discussed in the follow-up call, returning an error instead as there's no provision for NaN in JSON.